### PR TITLE
create Makefile for faster iteration

### PR DESCRIPTION
**Pull request by jdevries3133**
* makefile can automatically build the package, then navigate to the
  cpython repository to install the package and/or rebuild documentation

### CODE REVIEW

**Comment by jdevries3133** [comment]
For context, this PR has no associated issue, I created this little Makefile while I was tinkering with this theme, and I thought that it might be useful to others.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+CPYTHON_PATH               = ../cpython
+PACKAGE_ABS_PATH = $(shell pwd)/$(shell find dist/python-docs-theme-*.tar.gz)
+
+
+.PHONY: all
+all: install
```
Comment (hugovk) [inline Makefile:6]: Could we use the same target names as in the CPython, devguide and PEPs repo? Less confusion and more muscle memory :)

```suggestion
.PHONY: html
html: venv
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	cd $(CPYTHON_PATH)/Doc && \
+		make html
+
+
+.PHONY: install
+install:
```
Comment (hugovk) [inline Makefile:12]:
```suggestion
.PHONY: venv
venv:
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	python3 -m build
+	cd $(CPYTHON_PATH)/Doc && \
+		./venv/bin/pip install $(PACKAGE_ABS_PATH)
+
+.PHONY: help
+help:
+	@echo "all:		default rule; run the \`install\` rule, and also rebuild the cpython docs"
+	@echo "install: 	build the package, and install it into the virtual environment"
```
Comment (hugovk) [inline Makefile:20]:
```suggestion
	@echo "html:		default rule; run the \`venv\` rule, and also rebuild the CPython docs"
	@echo "venv: 		build the package, and install it into the virtual environment"
```

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -13,3 +13,12 @@
 - bump version (YYYY.MM.dev) in setup.py and python_docs_theme/theme.conf
 - Commit this last bump.
 - push and push the tag (``git push && git push --tags``)
+
+# Makefile Usage
```
Comment (hugovk) [inline CONTRIBUTING.rst:17]: Consistency with previous header
```suggestion
# Makefile usage
```

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -13,3 +13,12 @@
...
+
+This project includes a simple Makefile for syncing changes to the theme with
+the main cpython repository. Run ``make help`` for details on available rules.
+
+There is one configurable variable, ``CPYTHON_PATH``, which should be the path
+to the cpython repository on your machine. By default, it points to
```
Comment (hugovk) [inline CONTRIBUTING.rst:23]:
```suggestion
the main CPython repository. Run ``make help`` for details on available rules.

There is one configurable variable, ``CPYTHON_PATH``, which should be the path
to the CPython repository on your machine. By default, it points to
```

**Review by hugovk — CHANGES_REQUESTED** [review]
Thanks for the PR! A couple of little suggestions.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+.PHONY: all
+all: install
+	cd $(CPYTHON_PATH)/Doc && \
+		make html
+
+
+.PHONY: install
+install:
+	python3 -m build
```
Comment (hugovk) [inline Makefile:13]: And we should install `build` before using.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	cd $(CPYTHON_PATH)/Doc && \
+		./venv/bin/pip install $(PACKAGE_ABS_PATH)
```
Comment (hugovk) [inline Makefile:15]: And we need a `make venv` in here, as the CPython repo might not have a venv in its `Doc` dir yet (also align ampersands for readability):
```suggestion
	cd $(CPYTHON_PATH)/Doc \
		&& make venv \
		&& ./venv/bin/pip install $(PACKAGE_ABS_PATH)
```

**Review by hugovk — APPROVED** [review]
Thanks!

**Pull request merged**